File: app/src/main/java/com/example/application/FizzBuzzValueCommand.java
package com.example.application;

import com.example.domain.type.FizzBuzzType;

public class FizzBuzzValueCommand implements FizzBuzzCommand {
  private FizzBuzzType type;

  public FizzBuzzValueCommand(FizzBuzzType type) {
    this.type = type;
  }

  @Override
  public String execute(int number) {
    return type.generate(number).getValue();
  }
}

File: app/src/FizzBuzzValueCommand.php
<?php

namespace App;

class FizzBuzzValueCommand implements FizzBuzzCommand
{
    private FizzBuzzType $type;

    public function __construct(FizzBuzzType $type)
    {
        $this->type = $type;
    }

    public function execute(int $number): string
    {
        $value = $this->type->generate($number);
        return $value->getValue();
    }
}
